src/services/TechRecordsService.ts

```diff
@@ -232,27 +232,68 @@ class TechRecordsService {
           // systemNumber search should return a unique record
           throw new HTTPError(500, ERRORS.NO_UNIQUE_RECORD);
         }
-        const uniqueRecord = data[0];
+        const uniqueRecord = data[0];// call payload
         const oldTechRec = this.getTechRecordToArchive(uniqueRecord, statusCode);
-        const newRecord: any = _.cloneDeep(oldTechRec);
-        oldTechRec.statusCode = STATUS.ARCHIVED;
-        _.mergeWith(newRecord, techRecord.techRecord[0], this.arrayCustomizer);
-        this.setAuditDetails(newRecord, oldTechRec, msUserDetails);
-        populateFields(newRecord);
-        uniqueRecord.techRecord.push(newRecord);
-        return uniqueRecord;
+        if(!newStatusCode) {
+          const newRecord: ITechRecord = cloneDeep(oldTechRec);
+          mergeWith(newRecord, techRecord.techRecord[0], this.arrayCustomizer);
+          this.setAuditDetails(newRecord, oldTechRec, msUserDetails);
+          populateFields(newRecord);
+          uniqueRecord.techRecord.push(newRecord);
+          return uniqueRecord;
```

Comment:
there are 4-5 lines of code that are repeated 3 times, can be extracted to a function